### PR TITLE
Marketplace: save stripe api key

### PR DESCRIPTION
**Pull request by KellyAH**
See: #831 

Added logic for saving stripe api key on Marketplace.
Integrating with stripe via saving the stripe api key was more difficult than we thought. 😞 
Screen Shot 2022-12-21 at 7 21 06 PM

Co-authored-by: Zee Spencer <zspencer@users.noreply.github.com>
Co-authored-by: Ana <anaulin@users.noreply.github.com>
Co-authored-by: Naomi Quinones <naomiquinones@users.noreply.github.com>

TODO:

- [x] add unit tests

### CODE REVIEW

File: app/furniture/marketplace/marketplace.rb [hunk 1]
```diff
@@ -5,6 +5,14 @@ class Marketplace < FurniturePlacement
     has_many :products, inverse_of: :marketplace, dependent: :destroy
     has_many :carts, inverse_of: :marketplace, dependent: :destroy
 
+    def stripe_api_key=(key)
+      settings[:stripe_api_key] = key
```
Comment (anaulin) [inline app/furniture/marketplace/marketplace.rb:9]:
```suggestion
      settings["stripe_api_key"] = key
```

File: app/furniture/marketplace/marketplace.rb [hunk 2]
```diff
@@ -5,6 +5,14 @@ class Marketplace < FurniturePlacement
...
+    end
+
+    def stripe_api_key
+      settings[:stripe_api_key]
```
Comment (anaulin) [inline app/furniture/marketplace/marketplace.rb:13]:
```suggestion
      settings["stripe_api_key"]
```

**Review by anaulin — APPROVED** [review]
🗝️

File: app/furniture/marketplace/marketplaces/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<%= form_with model: marketplace.location do |marketplace_form| %>
+  <h2> Configure Marketplace</h2>
+  <fieldset>
+    <%= render "password_field", form: marketplace_form, attribute: :stripe_api_key %>
+    <%= marketplace.stripe_api_key %>
```
Comment (zspencer) [inline app/furniture/marketplace/marketplaces/_form.html.erb:5]:
```suggestion
```

File: app/furniture/marketplace/marketplace_policy.rb [hunk 4]
```diff
@@ -5,6 +5,10 @@ class MarketplacePolicy < ApplicationPolicy
     def show?
       true
     end
+
+    def update?
+      true
```
Comment (zspencer) [inline app/furniture/marketplace/marketplace_policy.rb:10]: Maybe this logic should be thought through a bit before merge :D

**Review by zspencer — APPROVED** [review]
I think we could merge as is; or we could flesh out the policy logic first. Dealers choice, @KellyAH !!!

**Pull request merged**